ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h

```diff
@@ -802,23 +808,24 @@ void RichardsMechanicsLocalAssembler<ShapeFunctionDisplacement,
 
         storage_p_a_p.noalias() +=
             N_p.transpose() * rho_LR * specific_storage_a_p * N_p * w;
-        storage_p_a_S.noalias() += N_p.transpose() * rho_LR *
-                                   specific_storage_a_S * (S_L - S_L_prev) /
-                                   dt * w;
+        if (p_cap_dot_ip != 0)  // prevent division by zero.
```

Comment:
If `p_cap_dot_ip` is very small, this condition would causes a problem.

Reply:
> If `p_cap_dot_ip` is very small, this condition would cause a problem.

Yeah. In the saturated range p_cap_dot_ip = -p_L_dot_ip, and the potentially small pressure rate goes along with S_L = S_L_prev. So this case should be fine, I guess.
In the unsaturated case, the ratio of both depends on the shape of the SWCC, but in many cases it should work.

@endJunction if none of the benchmarks nor our additional tests run into any problems of this kind, then I'd be fine with this for now and only address it if an actual problem occurs.

Reply:
So far I didn't come across a case where it would produce anything strange.